yb-voyager/cmd/exportData.go

```diff
@@ -221,45 +222,63 @@ func debeziumExportData(ctx context.Context, tableList []*sqlname.SourceName) er
 	if err != nil {
 		return fmt.Errorf("failed to start debezium: %w", err)
 	}
-	var status *dbzm.ExportStatus
-	exportingSnapshot := true
-	for exportingSnapshot {
-		status, err = debezium.GetExportStatus()
+
+	if liveMigration {
+		// monitor the debezium process and mode = snapshot/streaming
+		snapshotComplete := false
+		for debezium.IsRunning() {
+			if !snapshotComplete {
+				snapshotComplete, err = verifyAndHandleDebeziumSnapshotComplete(*debezium)
+				if err != nil {
+					utils.ErrExit("could not verify debezium snapshot status: %w", err)
+				}
+				if snapshotComplete {
+					utils.PrintAndLog("Streaming changes to a local queue file...")
+				}
+			}
+			time.Sleep(time.Second)
+		}
+		err = debezium.Error()
 		if err != nil {
-			return fmt.Errorf("failed to read export status: %w", err)
+			utils.ErrExit("debezium exited unexpectedly: %w\nOutput:%s", err, debezium.CombinedOutput())
 		}
-
-		if status != nil && exportingSnapshot && status.SnapshotExportIsComplete() {
-			exportingSnapshot = false
-			utils.PrintAndLog("Snapshot export is complete.")
-			createExportDataDoneFlag()
-			err = writeDataFileDescriptor(exportDir, status)
-			if err != nil {
-				return fmt.Errorf("failed to write data file descriptor: %w", err)
+	} else {
+		// offline migration.
+		// monitor the debezium process
+		for debezium.IsRunning() {
+			time.Sleep(time.Second)
+		}
+		err = debezium.Error()
+		if err != nil {
+			utils.ErrExit("debezium exited unexpectedly: %w\nOutput:%s", err, debezium.CombinedOutput())
+		} else {
+			// verify snapshot is complete.
+			snapshotComplete, err := verifyAndHandleDebeziumSnapshotComplete(*debezium)
+			if !snapshotComplete || err != nil {
+				utils.ErrExit("debezium exited unexpectedly: %w\nOutput:%s", err, debezium.CombinedOutput())
 			}
-			outputExportStatus(status)
 		}
-		time.Sleep(time.Second)
-	}
-	if err := debezium.Error(); err != nil {
-		return fmt.Errorf("debezium failed during initial snapshot phase: %w", err)
 	}
```

Comment:
```
tableNameToApxTotalRowCount, err := getTableNameToApxTotalRowCountMap(tableNames)
if err ...
debezium := dbzm.NewDebezium(config)
err = debezium.Start()
if err ...

progressTracker := NewProgressTracker(tableNameToApxTotalRowCount)

for debezium.IsRunning() {
	status, err = debezium.GetExportStatus()
	if err ...
	if status != nil {
		progressTracker.UpdateProgress(status)
	}
	time.Sleep(2 * time.Second)
}
status, err = debezium.GetExportStatus()
err = debezium.Error()
progressTracker.Done(status, err)
if err ...
```
@sanyamsinghal is working on adding progress-bar support and I have suggested him the above code snippet.
I would suggest to remove the related code changes from this PR. By moving everything related to "progress" in the "progress tracker", we no more need two separate branches for live/offline migration. Only requirement is to continuously report progress until debezium is running. When debezium is not running either show an error or final progress report.

Reply:
Let me catch up with sanyam on this. I was discussing the same with sanyam on friday about the progress-tracker taking the entire responsibility, but I'm not sure which component would handle switching from snapshot mode to streaming mode in live-migration(i.e. to create the flag, file descriptor, etc)? 

Does it make sense for the progress tracker to do that?